### PR TITLE
Fix permissions, based on scheme introduced by RabbitMQ 3.7.*

### LINKED ISSUE #703: Owner of /etc/rabbitmq
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: any
- Ruby: any
- Distribution: Gentoo
- Module version: Latest

## How to reproduce (e.g Puppet code you use)

Default code.

## What are you seeing

Wrong permissions on /etc/rabbitmq or NOT forced.

## What behaviour did you expect instead

Permissions of /etc/rabbitmq should be force to be owned by `$rabbitmq_user`. Because, it's required by : `/usr/sbin/rabbitmq-plugins enable rabbitmq_management` which creates `enabled_plugins` file in config dir under rabbitmq_user.

## Output log

```
 # /usr/sbin/rabbitmq-plugins enable rabbitmq_management
Error: {cannot_read_en
```

### PR DESCRIPTION
**Pull request by dhoppe**
Fixes #703

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -161,7 +161,7 @@
     content => template($config),
     owner   => '0',
     group   => $rabbitmq_group,
-    mode    => '0640',
+    mode    => '0644',
```
Comment (wyardley) [inline manifests/config.pp]: If the directories themselves are readable by the group, as are the files, I think these files can (and should?) stay 0640?

Comment (dhoppe) [reply manifests/config.pp]: I have my doubts, too. On the one hand, SSL certificates should not be readable by everyone, but on the other hand these defaults are given by the SGID bit via /etc/rabbitmq. 

Maybe this question can be answered by someone who has more experience with RabbitMQ?

Comment (wyardley) [reply manifests/config.pp]: What do you mean given by the sgid bit on the directory? You can have files with mode 0640 in a directory that’s world readable or executable w sgid bit (say 2755 or 2751). Sgid bit just affects default group ownership of files created in the directory afaik

**Comment by wyardley** [comment]
@dhoppe hope you don't mind - took a pass at a slightly narrower version of this in #835

**Pull request closed without merging**